UpcomingMovies/UpcomingMovies/Component/MovieComponent.cs

```diff
@@ -0,0 +1,111 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Threading.Tasks;
+using UpcomingMovies.Consts;
+using UpcomingMovies.DTO;
+using UpcomingMovies.Enums;
+using UpcomingMovies.Model;
+using UpcomingMovies.Parameter;
+using UpcomingMovies.Service;
+
+namespace UpcomingMovies.Component
+{
+    public class MovieComponent
+    {
+        IService _baseService;
+        public MovieComponent()
+        {
+
+        }
+        public MovieComponent(IService service)
+        {
+            _baseService = service;
+        }
+        public string Resource { get; set; }
+        internal async Task<UpcomingMovieModel> GetMovies(MovieParameter movieParameter)
+        {
+            _baseService = new BaseService(Resource);
+            var upComingMovieModel = new UpcomingMovieModel();
+            try
+            {
+                var response = await _baseService.Consume<MovieParameter, ResponseListDTO<List<MovieDTO>>>(movieParameter, HTTPMethodEnum.GET);
+                var movieModelList = response.results.Select(movie => new MovieModel
+                {
+                    ReleaseDate = string.IsNullOrEmpty(movie.release_date) ? "not available" : DateTime.Parse(movie.release_date).ToShortDateString(),
+                    Name = movie.title,
+                    OverView = BuildAbreviatedMovieOverView(movie.overview),
+                    Poster = BuildPosterUri(movie.poster_path),
+                    Score = movie.vote_average,
+                    Votes = movie.vote_count,
+                    Id = movie.id
+                }).ToList();
+                upComingMovieModel.Movies = movieModelList;
+                upComingMovieModel.Page = response.page;
+                upComingMovieModel.Total = response.total_pages;
+                upComingMovieModel.NextPage = response.page + 1;
+            }
+            catch (Exception ex)
+            {
+                upComingMovieModel.Error = true;
+                upComingMovieModel.ErrorMessage = ex.Message;
+            }
+            return upComingMovieModel;
+        }
+        internal async Task<MovieModel> GetMovie(MovieParameter movieParameter)
+        {
+            var movieresource = string.Format(MoviesApiResourcesConsts.MOVIE, movieParameter.Id);
+            movieParameter.Id = null;
+            _baseService = new BaseService(movieresource);
+            MovieModel movieModel = new MovieModel();
+            try
+            {
+                var response = await _baseService.Consume<MovieParameter, MovieDTO>(movieParameter, HTTPMethodEnum.GET);
+                var releaseDate = new DateTime();
+                DateTime.TryParse(response.release_date, out releaseDate);
+                var releaseDateFormatted = releaseDate == DateTime.MinValue ? string.Empty : DateTime.Parse(response.release_date).ToShortDateString();
```

Comment:
You parse `response.release_date` twice in a row.

Reply:
Wow, didn't saw this when I was coding, totally unnecessary.